pkg/build/strategies/sti/postexecutorstep.go

```diff
@@ -450,11 +422,12 @@ func createLabelsForResultingImage(builder *STI, docker dockerpkg.Docker, baseIm
 	}
 
 	configLabels := builder.config.Labels
+	newLabels := builder.newLabels
 
-	return mergeLabels(configLabels, generatedLabels, existingLabels)
+	return mergeLabels(configLabels, generatedLabels, existingLabels, newLabels)
 }
 
-func mergeLabels(configLabels, generatedLabels, existingLabels map[string]string) map[string]string {
+func mergeLabels(configLabels, generatedLabels, existingLabels map[string]string, newLabels map[string]string) map[string]string {
```

Comment:
You should be able to re-factor this to use a variadic function.
https://gobyexample.com/variadic-functions

```
func mergeLabels(labels... map[string]string) map[string]string {
 	mergedLabels := map[string]string{}
 	
	for _, labelMap := range labels {
		for k, v := range labelMap {
			mergedLabels[k] = v
		}
	}
 	return mergedLabels
 }
```